### PR TITLE
Migrate the SlowCheetah extension to support Dev17

### PR DESCRIPTION
**Pull request by cuiximu**
Migrate the SlowCheetah extension to support Dev17

1.Create shared projects and move core files to shared projects.
2. Create a new Dev17 VSIX project.

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -6,5 +6,6 @@
   <packageSources>
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
```
Comment (chzuluag) [inline nuget.config:7]: We need to consolidate a single PackageSource here.

File: nuget.config [hunk 2]
```diff
@@ -6,5 +6,6 @@
...
     <add key="vs-devcore" value="https://www.myget.org/F/vs-devcore/api/v3/index.json" />
+	  <add key="vs-impl" value="https://devdiv.pkgs.visualstudio.com/_packaging/vs-impl/nuget/v3/index.json" />
```
Comment (AArnott) [inline nuget.config:9]: nit: mixed indentation

File: nuget.config [hunk 3]
```diff
@@ -6,5 +6,6 @@
   <packageSources>
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
     <add key="vs-devcore" value="https://www.myget.org/F/vs-devcore/api/v3/index.json" />
```
Comment (AArnott) [inline nuget.config:8]: The myget feed is gone. You should be able to remove this.

Comment (cuiximu) [reply nuget.config]: Remove.

File: nuget.config (hunk 2 above)
Comment (cuiximu) [reply nuget.config]: Update.

File: nuget.config [hunk 4]
```diff
@@ -1,11 +1,10 @@
...
 <configuration>
-  <config>
-    <add key="repositorypath" value="packages" />
-  </config>
   <packageSources>
-    <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
-    <add key="vs-devcore" value="https://www.myget.org/F/vs-devcore/api/v3/index.json" />
-	  <add key="vs-impl" value="https://devdiv.pkgs.visualstudio.com/_packaging/vs-impl/nuget/v3/index.json" />
+    <clear />
+	<add key="vs-impl" value="https://devdiv.pkgs.visualstudio.com/_packaging/vs-impl/nuget/v3/index.json" />
```
Comment (AArnott) [inline nuget.config:5]: This feed requires authentication, which essentially blocks use of this repository to OSS developers. Can we use a public feed instead?

Comment (cuiximu) [reply nuget.config]: Done

File: nuget.config (hunk 3 above)
Comment (adrianvmsft) [reply nuget.config]: This was fixed in #234

**Comment by terryt2000** [comment]
What is holding up this pr?

**Comment by adrianvmsft** [comment]
> What is holding up this pr?

Current approach has some limitations - it increases complexity and you can't build the entire solution in VS (due to a limitation in VSSDK).
We are trying a simplified approach in #240. That means we would use different branches to target VS2022 vs. older versions of VS.

**Comment by terryt2000** [comment]
ah...thanks for the response.  I'm having to go back to using VS 2019 to debug  one of my projects that I use SlowCheetah to transform the configs for different customers

**Comment by adrianvmsft** [comment]
@terryt2000 We've just published the SlowCheetah extension for VS 2022 in case you would like to give it a try. You can find it [here](https://marketplace.visualstudio.com/items?itemName=vscps.SlowCheetah-XMLTransforms-VS2022)

**Comment by ImGonaRot** [comment]
@adrianvmsft Thanks, this is working great. Can you also close out the linked item #229

**Pull request closed without merging**